# **Análisis Exploratorio de Datos (EDA)**

En este cuaderno, nuestro objetivo principal es llevar a cabo un análisis exploratorio de los datos **(EDA)** relacionados con los siniestros viales en la Ciudad de Buenos Aires. El propósito de este análisis es obtener información descriptiva y relevante sobre este fenómeno particular. A través del EDA, buscamos identificar las características clave, las tendencias, las relaciones entre variables y cualquier anomalía presente en los datos.

> **Nota:** Si no cargan los graficos puede acceder al notebook desde Google Colab con el siguiente enlace: [EDA.ipynb](https://colab.research.google.com/drive/1cF_Yhlbimndm-At0CyfVEHSn1Hha07tw?usp=sharing)


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import plotly.graph_objects as go
import plotly.express as px

In [322]:
Homicidios = pd.read_csv('/content/drive/MyDrive/Homicidios/Homicidios.csv')

## **Histórico de Víctimas Fatales**
Para comprender la evolución temporal de las víctimas fatales en accidentes, realizamos un análisis del histórico de víctimas fatales a lo largo de los años. Utilizamos los datos del dataset Homicidios y nos enfocamos en la variable 'fecha_hora' para determinar la cantidad de víctimas fatales registradas en cada año.

A continuación, presentamos el gráfico de barras interactivo que muestra la cantidad de víctimas fatales por año:

In [323]:
# Convertir la columna 'fecha_hora' a tipo datetime
Homicidios['fecha_hora'] = pd.to_datetime(Homicidios['fecha_hora'])

# Calcular la cantidad de víctimas fatales por año
años = Homicidios['fecha_hora'].dt.year.value_counts().sort_index()

# Crear el gráfico de barras con Plotly y asignar diferentes colores
fig = px.bar(x=años.index, y=años.values,
             labels={'x': 'Año', 'y': 'Cantidad de víctimas fatales'},
             title='<b>Histórico de Víctimas Fatales</b>',  # Título en negrita
             color=años.index,  # Asignar diferentes colores basados en el año
             color_discrete_sequence=px.colors.qualitative.Pastel)  # Usar una paleta de colores pastel

# Centrar el título
fig.update_layout(title_x=0.5)

# Mostrar el gráfico
fig.show()

Durante el año 2020, se implementó medidas estrictas de confinamiento en respuesta a la pandemia de COVID-19. Estas medidas resultaron en una notable disminución en la movilidad vehicular. Como consecuencia directa, se observó una reducción significativa en el número de siniestros viales con víctimas fatales en comparación con años anteriores.

Sin embargo, a medida que se comenzó a flexibilizar las restricciones y a reactivar gradualmente la actividad económica y social en 2021, se registró un aumento en la circulación vehicular. Este incremento en la actividad vial estuvo acompañado de un repunte en el número de siniestros viales con víctimas fatales. Durante el año 2021, se reportaron un total de 97 siniestros de esta índole segun los registros de Buenos Aires Caba, cifra superior a los 81 ocurridos en el año anterior.

Es importante destacar que, a pesar del aumento en el número de siniestros fatales en 2021, estos aún se mantienen por debajo de los niveles observados antes de la pandemia. En 2019, año previo al brote de COVID-19, se reportaron 104 siniestros viales con víctimas fatales. Este contexto sugiere que si bien ha habido una recuperación en la actividad vial, aún no se ha alcanzado el nivel previo a la pandemia.

In [324]:
# Calcular la cantidad de víctimas fatales por mes
meses = Homicidios['fecha_hora'].dt.month.value_counts().sort_index()

# Calcular el porcentaje de víctimas fatales por mes
total_victimas = meses.sum()
porcentaje_meses = (meses / total_victimas * 100).astype(int) # Convertir el porcentaje a entero

# Crear un DataFrame con los datos
df_meses = pd.DataFrame({'Meses': meses.index, 'Cantidad de víctimas': meses.values, 'Porcentaje de víctimas': porcentaje_meses.values})

# Definir el nombre de los meses
nombres_meses = ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo', 'Junio',
                 'Julio', 'Agosto', 'Septiembre', 'Octubre', 'Noviembre', 'Diciembre']

# Asignar colores diferentes a cada barra
colores = px.colors.qualitative.Plotly

# Crear el gráfico de barras horizontales
fig = px.bar(df_meses, y=nombres_meses, x=['Cantidad de víctimas', 'Porcentaje de víctimas'],
             labels={'value': 'Cantidad de víctimas', 'variable': 'Medida', 'Meses': 'Meses'},
             title='Siniestros viales según mes de ocurrencia',
             color='Meses', color_discrete_sequence=colores,
             text=df_meses.apply(lambda row: f"{row['Cantidad de víctimas']} ({row['Porcentaje de víctimas']}%)", axis=1),
             orientation='h', # Orientación horizontal de las barras
             category_orders={"Meses": nombres_meses}, # Orden de los meses en el eje vertical
             width=800, height=600) # Ajuste del tamaño del gráfico

# Establecer el formato del título
fig.update_layout(title=dict(text='<b>Siniestros viales según mes de ocurrencia</b>', x=0.5))

# Establecer los nombres de los ejes
fig.update_layout(
    xaxis_title="Cantidad de víctimas",
    yaxis_title="Meses",
)

# Mostrar el gráfico
fig.update_traces(textposition='outside') # Posición del texto afuera de las barras
fig.show()

### **Análisis del Histórico de Víctimas Fatales**

El análisis revela una variación notable en el número de víctimas fatales a lo largo del año, destacando un pico significativo en diciembre. A continuación, se detalla el número de víctimas fatales para cada mes:

- **Enero:** 62 víctimas fatales
- **Febrero:** 59 víctimas fatales
- **Marzo:** 56 víctimas fatales
- **Abril:** 52 víctimas fatales
- **Mayo:** 60 víctimas fatales
- **Junio:** 58 víctimas fatales
- **Julio:** 51 víctimas fatales
- **Agosto:** 67 víctimas fatales
- **Septiembre:** 51 víctimas fatales
- **Octubre:** 52 víctimas fatales
- **Noviembre:** 68 víctimas fatales
- **Diciembre:** 81 víctimas fatales

Destaca el mes de diciembre con un número considerablemente mayor de víctimas fatales en comparación con otros meses, representando el 11% del total anual. Este aumento puede atribuirse a una serie de factores, como el incremento del tráfico debido a las festividades de fin de año, el aumento del consumo de alcohol durante las celebraciones, condiciones climáticas adversas o una disminución de la precaución general por parte de los conductores y peatones debido al ambiente festivo.

Este análisis resalta la importancia de implementar medidas de seguridad vial y campañas de concientización, especialmente durante los períodos festivos, con el objetivo de reducir el número de víctimas fatales en las carreteras.


In [325]:
# Calcular la cantidad de víctimas fatales por día de la semana
dias_semana = Homicidios['fecha_hora'].dt.dayofweek.value_counts().sort_index()

# Calcular el total de víctimas
total_victimas = dias_semana.sum()

# Calcular el porcentaje de víctimas por día de la semana
porcentaje_dias_semana = (dias_semana / total_victimas * 100)

# Crear un DataFrame con los datos
df_dias_semana = pd.DataFrame({'Día de la semana': ['Lunes', 'Martes', 'Miércoles', 'Jueves', 'Viernes', 'Sábado', 'Domingo'],
                               'Cantidad de víctimas': dias_semana.values,
                               'Porcentaje de víctimas': porcentaje_dias_semana.values})

# Crear el gráfico de barras
fig = px.bar(df_dias_semana, x='Día de la semana', y='Cantidad de víctimas',
             labels={'Cantidad de víctimas': 'Cantidad de víctimas', 'Día de la semana': 'Día de la semana'},
             color='Día de la semana', color_discrete_sequence=px.colors.qualitative.Pastel,
             text=df_dias_semana.apply(lambda row: f"{row['Cantidad de víctimas']} ({row['Porcentaje de víctimas']:.2f}%)", axis=1))

# Configurar el diseño del gráfico
fig.update_traces(textposition='outside', textfont_size=12)

# Establecer el formato del título
fig.update_layout(title=dict(text='<b>Siniestros viales según día de ocurrencia</b>', x=0.5))

# Mostrar el gráfico
fig.show()

Los días sábado y lunes registraron la misma cantidad de siniestros viales con víctimas fatales, con un total de **107** casos cada uno, lo que representa aproximadamente el 14.92% del total. Le siguen los días domingo, con 105 casos (14.64%), y viernes, con 102 casos (14.23%).

Un análisis revela que los fines de semana concentran la mayor cantidad de siniestros viales con víctimas fatales. Específicamente, los días sábado y domingo son los más críticos en términos de accidentes mortales en la vía pública, seguidos de cerca por los lunes. Este patrón puede estar relacionado con diversos factores, como el aumento de la actividad recreativa y el consumo de alcohol durante los fines de semana, lo que podría contribuir a una mayor incidencia de accidentes.

In [326]:
# Calcular la cantidad de víctimas fatales por hora del día
horas = Homicidios['fecha_hora'].dt.hour.value_counts().sort_index()

# Crear un DataFrame con los datos
df_lineas = pd.DataFrame({'Hora del día': horas.index, 'Cantidad de víctimas': horas.values})

# Generar el gráfico de líneas
fig = px.line(df_lineas, x='Hora del día', y='Cantidad de víctimas',
              labels={'Hora del día': 'Hora del día', 'Cantidad de víctimas': 'Cantidad de víctimas'},
              title='Siniestros viales según hora de ocurrencia')

# Configurar el diseño del gráfico
fig.update_layout(title=dict(text='<b>Siniestros viales según hora de ocurrencia</b>', x=0.5))

# Agregar una línea roja
fig.add_trace(go.Scatter(x=df_lineas['Hora del día'], y=df_lineas['Cantidad de víctimas'],
                         mode='lines', line=dict(color='red'), name='Tendencia'))

# Mostrar el gráfico
fig.show()

El análisis de los datos revela que los períodos de mayor siniestralidad vial se concentran en las franjas horarias de la mañana temprano, de 5:00 a 7:59 horas, y en la tarde, de 16:00 a 19:59 horas. Durante estas horas, se registraron 125 y 124 siniestros viales, respectivamente, lo que indica una alta incidencia de accidentes durante esos momentos del día.

Este patrón sugiere que los momentos de mayor actividad vial, como la hora pico de la mañana y la tarde, están asociados con un mayor riesgo de siniestros viales. Es posible que factores como el tráfico congestionado, la fatiga del conductor y las condiciones de iluminación desfavorables contribuyan a esta tendencia.

El reporte detallado de víctimas fatales por hora del día muestra que, en general, la cantidad de siniestros tiende a disminuir durante la noche y la madrugada, alcanzando su punto más bajo entre la medianoche y las 4:59 horas.

Este análisis resalta la importancia de la conciencia y la precaución al volante, especialmente durante los períodos de mayor riesgo identificados, con el fin de reducir la incidencia de siniestros viales y proteger la seguridad de todos los usuarios de la vía.

In [327]:
# Calcular la cantidad de víctimas fatales por tipo de vehículo
tipo_vehiculo = Homicidios['victima'].value_counts()

# Crear un DataFrame con los datos
df_tipo_vehiculo = pd.DataFrame({'Tipo de vehículo': tipo_vehiculo.index, 'Cantidad de víctimas': tipo_vehiculo.values})

# Crear el gráfico de barras horizontal con Plotly Express
fig = px.bar(df_tipo_vehiculo,
             x='Cantidad de víctimas',
             y='Tipo de vehículo',
             labels={'Cantidad de víctimas': 'Cantidad de víctimas', 'Tipo de vehículo': 'Tipo de vehículo'},
             title='Víctimas Fatales según Tipo de Vehículo',
             color='Tipo de vehículo',
             color_discrete_sequence=px.colors.qualitative.Plotly)

# Personalizar el diseño del gráfico
fig.update_layout(title=dict(text='<b>Víctimas Fatales según Tipo de Vehículo</b>', x=0.5),
                  xaxis_title='Cantidad de víctimas fatales',
                  yaxis_title='Tipo de vehículo',
                  plot_bgcolor='white',
                  paper_bgcolor='white',
                  font=dict(color='black'))

# Mostrar el gráfico
fig.show()

Durante el período de 2016 a 2021, se observa que las motocicletas son el tipo de vehículo más comúnmente implicado en siniestros viales con víctimas fatales, con un total de 311 casos, lo que representa aproximadamente el 60% del total. Los peatones ocupan el segundo lugar en términos de incidencia, con 268 víctimas fatales (37%), seguidos por los automóviles con 94 casos (13%). Las bicicletas contribuyen con 29 casos (4%) y los vehículos de carga y pasajeros tienen una participación menor, con 7 (1%) y 5 (1%) casos respectivamente. También se registraron 3 casos (menos del 1%) donde el tipo de vehículo no fue especificado en el informe, pero estos valores fueron imputados por la moda de los valores.

In [328]:
# Calcular el conteo de ocurrencias de cada tipo de acusado
conteo_acusados = Homicidios['acusado'].value_counts()

# Crear un DataFrame con los datos para el gráfico
df_grafico = pd.DataFrame({'Acusado': conteo_acusados.index, 'Cantidad de acusados': conteo_acusados.values})

# Ordenar el DataFrame por la cantidad de acusados de manera descendente
df_grafico = df_grafico.sort_values(by='Cantidad de acusados', ascending=False)

# Colores para las barras
colores = px.colors.qualitative.Pastel

# Crear el gráfico de barras horizontal
fig = px.bar(df_grafico,
             x='Cantidad de acusados',
             y='Acusado',
             orientation='h',  # Orientación horizontal
             title='Acusados por Tipo de Vehículo',
             labels={'Cantidad de acusados': 'Cantidad de acusados', 'Acusado': 'Tipo de vehículo'},
             text='Cantidad de acusados',  # Mostrar etiquetas de valores en las barras
             color='Acusado',  # Asignar colores diferentes a cada barra
             color_discrete_map={acu: col for acu, col in zip(df_grafico['Acusado'], colores)})  # Mapear colores

# Configurar el diseño del gráfico
fig.update_layout(xaxis_title='Cantidad de acusados',  # Título del eje x
                  yaxis_title='Tipo de vehículo',  # Título del eje y
                  title={'text': '<b>Acusados por Tipo de Vehículo</b>', 'x': 0.5},  # Título centrado y en negrita
                  plot_bgcolor='rgba(0,0,0,0)',  # Color de fondo del gráfico
                  paper_bgcolor='rgb(255,255,255)')  # Color de fondo del papel

# Mostrar el gráfico
fig.show()

En relación con el tipo de vehículo involucrado en las acusaciones de siniestros viales, destaca que el 29% de los acusados estaban al mando de un vehículo tipo AUTO, con un total de 231 acusaciones. Le sigue en frecuencia el tipo de vehículo PASAJEROS, con 178 acusados (25%), y CARGAS, con 150 acusados (21%).

Es interesante destacar que, a pesar de la alta siniestralidad asociada a las motocicletas, estas representan solo el 8% de las acusaciones, con un total de 58. Esto sugiere que, aunque las motocicletas puedan estar involucradas en numerosos accidentes, rara vez el conductor de una motocicleta resulta acusado del incidente.

Estos hallazgos subrayan la importancia de analizar no solo la frecuencia de los tipos de vehículos involucrados en siniestros viales, sino también su relación con las acusaciones resultantes, lo que puede proporcionar información valiosa para la implementación de medidas de prevención y seguridad vial.

In [329]:
# Orden de los rangos de edad
orden_rangos = ['0-17', '18-29', '30-44', '45-64', '65+']

# Calcular el conteo de víctimas por rango de edad
conteo_edades = Homicidios['categoria_edad'].value_counts().reindex(orden_rangos)

# Crear un DataFrame con los datos para el gráfico
df_grafico = pd.DataFrame({'Rango Etario': conteo_edades.index, 'Cantidad de víctimas': conteo_edades.values})

# Crear el gráfico de barras
fig = px.bar(df_grafico, x='Cantidad de víctimas', y='Rango Etario',
             orientation='h',  # Orientación horizontal
             title='Víctimas Fatales por Rango Etario',
             labels={'Cantidad de víctimas': 'Cantidad de víctimas', 'Rango Etario': 'Rango Etario'},
             text='Cantidad de víctimas',  # Mostrar etiquetas de valores en las barras
             color='Rango Etario',  # Asignar colores diferentes a cada barra
             color_discrete_sequence=px.colors.qualitative.Pastel)  # Colores para las barras

# Configurar el diseño del gráfico
fig.update_layout(xaxis_title='Cantidad de víctimas',  # Título del eje x
                  yaxis_title='Rango Etario',  # Título del eje y
                  title={'text': '<b>Víctimas Fatales por Rango Etario</b>', 'x': 0.5},  # Título centrado y en negrita
                  plot_bgcolor='rgba(0,0,0,0)',  # Color de fondo del gráfico
                  paper_bgcolor='rgb(255,255,255)',  # Color de fondo del papel
                  bargap=0.2)  # Espacio entre las barras

# Mostrar el gráfico
fig.show()

Basándonos en el análisis de los datos, se puede destacar una disparidad significativa en los perfiles de las víctimas fatales en relación con su rango de edad. Se observa claramente una predominancia de víctimas en el rango de edad de 18 a 44 años, con un total de 399 casos, lo que representa aproximadamente el 57% del total de víctimas fatales registradas. Esta cifra es notablemente superior a las cifras de otros grupos de edad. Por ejemplo, el grupo de edad de 30 a 44 años presenta el mayor número de víctimas fatales con 254 casos, seguido del grupo de edad de 18 a 29 años con 198 casos. Este análisis sugiere que los adultos jóvenes, especialmente aquellos en el rango de edad de 18 a 44 años, son el grupo demográfico más afectado por los siniestros viales en términos de fatalidades.

In [330]:
# Gráfico de barras
fig = px.bar(Homicidios, y="sexo", orientation='h',
             title='<b>Víctimas Fatales por Sexo</b>',
             labels={'sexo': 'Cantidad de víctimas', 'index': 'Sexo'},
             color_discrete_sequence=px.colors.qualitative.Pastel)
fig.update_traces(textposition='inside', textfont_size=12)

# Centrar el título del gráfico de barras
fig.update_layout(title=dict(text='<b>Víctimas Fatales por Sexo</b>', x=0.5),
                  xaxis_title="Cantidad de víctimas", yaxis_title="Sexo")

# Gráfico de pastel
fig_pie = go.Figure()

sexo = Homicidios['sexo'].value_counts()

fig_pie.add_trace(go.Pie(labels=sexo.index, values=sexo,
                          textinfo='percent+label',
                          hole=0.7,
                          marker_colors=['#ff9999', '#66b3ff'],
                          hoverinfo='label+percent',
                          textposition='inside'))

# Centrar el título del gráfico de pastel
fig_pie.update_layout(title=dict(text='<b>% de Víctimas Fatales por Sexo</b>', x=0.5),
                      annotations=[dict(text='Sexo', x=0.5, y=0.5, font_size=20, showarrow=False)])

# Mostrar los gráficos
fig.show()
fig_pie.show()


Claramente se observa una disparidad significativa en cuanto al sexo de las víctimas, donde la gran mayoría, casi 8 de cada 10, corresponden al sexo masculino, totalizando 545 víctimas. Esto representa más del 75% del total de víctimas fatales registradas. En contraste, las víctimas femeninas alcanzan un total de 166, representando aproximadamente el 23.2% del total. Esta diferencia resalta la importancia de considerar el género al analizar la incidencia de víctimas fatales en siniestros viales.

In [331]:
# Crear el gráfico utilizando Plotly
fig = px.histogram(Homicidios, x='sexo', color='categoria_edad', barmode='group',
                   category_orders={"categoria_edad": ['0-17', '18-29', '30-44', '45-64', '65+']},
                   labels={'sexo': 'Sexo', 'categoria_edad': 'Rango Etario', 'cantidad_de_victimas': 'Cantidad de víctimas'},
                   title='<b>Víctimas Fatales por Sexo</b>',
                   template='plotly_white',  # Cambiar al tema blanco
                   color_discrete_sequence=px.colors.qualitative.Set1)  # Utilizar una paleta de colores profesional

# Renombrar la etiqueta del eje x
fig.update_xaxes(title_text='Cantidad de víctimas')

# Centrar el título del gráfico de barras y ponerlo en negrilla
fig.update_layout(title=dict(text='<b>Víctimas Fatales por Sexo</b>', x=0.5),
                  xaxis_title="Sexo", yaxis_title="Cantidad de víctimas")

# Mostrar el gráfico
fig.show()

En el análisis de las víctimas según su categoría de edad, se destaca una vez más la disparidad en cuanto al sexo de las víctimas, revelando una predominancia de víctimas masculinas en los rangos de edad más jóvenes, especialmente entre 18 y 45 años. Por el contrario, aunque el número de víctimas femeninas es significativamente menor en comparación con las masculinas, se observa una mayor concentración de víctimas femeninas en los grupos de edad más avanzada. Este patrón sugiere posibles diferencias en los factores de riesgo y en la exposición a situaciones peligrosas según el sexo y la edad de las víctimas involucradas en siniestros viales.

In [332]:
# Conteo de víctimas según el tipo de vía
encrucijada = Homicidios['encrucijada'].value_counts()

# Gráfico de donas para el porcentaje de víctimas en encrucijadas
fig_donut = go.Figure(data=[go.Pie(labels=['No', 'Sí'],
                                   values=encrucijada.values,
                                   hole=0.7,
                                   textinfo='percent+label',
                                   marker=dict(colors=['#1f77b4', '#ff7f0e']),
                                   name='Porcentaje de Víctimas en Encrucijadas')])
fig_donut.update_layout(title='<b>Porcentaje de Víctimas en Encrucijadas</b>',
                        title_x=0.5)  # Centrar el título horizontalmente

# Gráfico de barras para la cantidad de víctimas en encrucijadas
fig_bar = go.Figure(data=[go.Bar(x=['No', 'Sí'],
                                 y=encrucijada.values,
                                 marker=dict(color=['#1f77b4', '#ff7f0e']))])
fig_bar.update_layout(title='<b>Cantidad de Víctimas en Encrucijadas</b>',
                      xaxis_title='Encrucijada',
                      yaxis_title='Cantidad de víctimas',
                      title_x=0.5)  # Centrar el título horizontalmente

# Mostrar los gráficos
fig_donut.show()
fig_bar.show()

In [333]:
# Definir una lista de colores para cada barra
colores = ['rgb(31, 119, 180)', 'rgb(255, 127, 14)', 'rgb(44, 160, 44)', 'rgb(214, 39, 40)', 'rgb(148, 103, 189)']

# Gráfico de barras para la cantidad de víctimas por tipo de vía
vias = Homicidios['tipo_de_calle'].value_counts()
fig_bar = go.Figure(data=[go.Bar(x=vias.index,
                                 y=vias.values,
                                 marker_color=colores)])  # Asignar la lista de colores
fig_bar.update_layout(title='<b>Víctimas Fatales por Tipo de Vía</b>',
                      xaxis_title='Tipo de Vía',
                      yaxis_title='Cantidad de víctimas',
                      title_x=0.5)  # Centrar el título horizontalmente
fig_bar.show()

In [334]:
# Definir los colores personalizados para el gráfico de dona
colors = ['rgb(31, 119, 180)', 'rgb(255, 127, 14)', 'rgb(44, 160, 44)', 'rgb(214, 39, 40)', 'rgb(148, 103, 189)', 'rgb(140, 86, 75)', 'rgb(227, 119, 194)', 'rgb(127, 127, 127)', 'rgb(188, 189, 34)']

# Gráfico de dona para el porcentaje de víctimas por tipo de vía
fig_donut = go.Figure(data=[go.Pie(labels=vias.index,
                                   values=vias.values,
                                   hole=0.7,
                                   textinfo='percent+label',
                                   marker=dict(colors=colors))])
fig_donut.update_layout(title='<b>% de Víctimas Fatales por Tipo de Vía</b>',
                        title_x=0.5)  # Centrar el título horizontalmente
fig_donut.show()

Se puede observar que la cantidad de víctimas en encrucijadas es considerablemente alta, con un total de 540 víctimas en aquellas que no presentan encrucijadas y 177 víctimas en aquellas que sí las presentan. Este análisis resalta la importancia de implementar medidas de seguridad específicas en las intersecciones para reducir la cantidad de siniestros viales con consecuencias fatales.

También se aprecia que la mayoría de los siniestros viales con consecuencias fatales tienen lugar en avenidas, representando un total de 442 incidentes, lo que corresponde al 62% del total. Específicamente, la gran mayoría de estos eventos, alrededor del 75%, ocurren en encrucijadas, lo que significa que 7 de cada 10 siniestros tienen lugar en intersecciones de calles.

El análisis revela que, aunque las avenidas son los lugares más comunes para este tipo de eventos, las encrucijadas dentro de estas vías presentan un riesgo significativamente mayor. Este hallazgo resalta la importancia de implementar medidas de seguridad específicas en las intersecciones para reducir la incidencia de siniestros viales fatales.

In [335]:
# Crear el gráfico de caja para las edades de cada tipo de vehículo
fig = px.box(Homicidios, x='victima', y='edad', color='victima')

# Configurar el diseño del gráfico y el título centrado y en negrilla
fig.update_layout(title='<b>Distribución de Edades por Tipo de Vehículo</b>',
                  xaxis_title='Tipo de Vehículo',
                  yaxis_title='Edad',
                  title_x=0.5)  # Centrar el título horizontalmente

# Mostrar el gráfico
fig.show()

Se pudo concluir las siguientes observaciones basadas en el gráfico Distribucción de Edades por Tipo de vehículo.

**AUTO:** El grupo de víctimas fatales en vehículos automotores tiene una media de edad de aproximadamente 40 años, con una dispersión moderada en torno a esta medida central. Las edades oscilan desde 1 hasta 78 años, con la mayoría de las víctimas concentradas entre los 28 y 50 años, como lo indica el rango intercuartílico (25% al 75%).

**BICICLETA:** Las víctimas fatales involucradas en accidentes con bicicletas tienen una edad media cercana a los 40 años, con una dispersión algo mayor que en el caso de los automóviles. Las edades varían desde los 5 hasta los 86 años, y la mayoría de las víctimas se encuentran entre los 26 y 49 años, según el rango intercuartílico.

**CARGAS:** En el caso de los accidentes que involucran vehículos de carga, la media de edad es alrededor de 41 años, con una dispersión menor en comparación con otros tipos de vehículos. La mayoría de las víctimas tienen entre 32 y 45 años, según el rango intercuartílico.

**MOTO:** Las víctimas fatales relacionadas con accidentes de motocicleta tienden a ser más jóvenes, con una media de edad de aproximadamente 33 años y una dispersión moderada en torno a esta medida. Las edades oscilan entre 16 y 84 años, y la mayoría de las víctimas se encuentran entre los 24 y 37 años, como lo indica el rango intercuartílico.

**MOVIL:** Aunque el grupo de víctimas fatales en vehículos móviles es pequeño, la edad media es de alrededor de 36 años, con una dispersión moderada en torno a esta medida central. Las edades van desde los 24 hasta los 55 años, y la mayoría de las víctimas se encuentran entre los 27 y 42 años, según el rango intercuartílico.

**PASAJEROS:** Las víctimas fatales que eran pasajeros tienen una edad media de aproximadamente 62 años, con una dispersión considerable en torno a esta medida central. Las edades varían desde los 20 hasta los 81 años, y la mayoría de las víctimas se encuentran entre los 61 y 79 años, según el rango intercuartílico.

**PEATON:** El grupo de víctimas fatales que eran peatones muestra una amplia variabilidad en las edades, con una media de alrededor de 53 años y una dispersión considerable en torno a esta medida central. Las edades oscilan desde 1 hasta 95 años, y la mayoría de las víctimas se encuentran entre los 35 y 73 años, como lo indica el rango intercuartílico.

Estas estadísticas descriptivas proporcionan una visión general de las edades de las víctimas fatales en diferentes escenarios de accidentes de tránsito, lo que puede ser útil para comprender mejor los factores de riesgo y orientar las estrategias de prevención.

La distribución de edades no es simétrica para la mayoría de los tipos de vehículos.
Los valores atípicos pueden indicar la presencia de subgrupos con características diferentes a la población general.

Existe una gran variabilidad en la edad de las víctimas de accidentes de tráfico, con diferencias significativas entre los diferentes tipos de vehículos.
Los motociclistas son el grupo más vulnerable en términos de edad, con la menor edad media y la mayor variabilidad.
Los pasajeros son el grupo con la edad media más alta, lo que puede estar relacionado con la mayor frecuencia de viajes por motivos de salud o trabajo.
Se recomienda realizar análisis adicionales para identificar los factores que influyen en la edad de las víctimas de accidentes de tráfico.

In [336]:
# Crear el gráfico de violín
fig = px.violin(Homicidios, x='victima', y='edad', box=True, points='all', color='victima')

# Configurar el diseño del gráfico
fig.update_layout(title='<b>Distribución de Edades de las Víctimas por Tipo de Vehículo</b>',
                  xaxis_title='Tipo de Vehículo',
                  yaxis_title='Edad',
                  title_x=0.5)  # Centrar el título horizontalmente

# Mostrar el gráfico
fig.show()

Según el gráfico de distribución de edades por tipo de vehículo, se confirma lo observado previamente. Hay una notable variabilidad en las edades de las víctimas fatales, dependiendo del vehículo que ocupaban en el momento del incidente. La distribución de edades de las víctimas peatones es la más amplia, abarcando todas las franjas etarias. En contraste, la distribución de edades de las víctimas que eran pasajeros tiende a concentrarse en edades adultas, mientras que las víctimas más jóvenes suelen estar involucradas en accidentes relacionados con motocicletas.

Al analizar las estadísticas descriptivas por tipo de vehículo, se observa que las edades promedio varían significativamente entre los diferentes tipos. Por ejemplo, las víctimas que viajaban en vehículos de carga y automóviles tienden a tener una edad promedio cercana a los 40 años, mientras que las víctimas de accidentes de tránsito que involucran motocicletas son más jóvenes, con una edad promedio de alrededor de 33 años. Por otro lado, los peatones y pasajeros presentan una mayor dispersión en sus edades, como lo indica la mayor desviación estándar en sus estadísticas descriptivas.

In [337]:
# Crear el gráfico de barras horizontal con colores personalizados
fig = px.bar(Homicidios, y='rol', orientation='h',
             title='<b>Víctimas Fatales por Rol</b>',  # Título en negrita
             labels={'rol': 'Rol', 'index': 'Cantidad de víctimas'},
             category_orders={'rol': Homicidios['rol'].value_counts().index},
             color='rol',  # Colorear por tipo de rol
             color_discrete_sequence=px.colors.qualitative.Set2)  # Colores personalizados

# Centrar el título horizontalmente
fig.update_layout(title_x=0.5)

# Cambiar el título del eje y
fig.update_layout(yaxis_title='Rol')

# Cambiar el título del eje x
fig.update_layout(xaxis_title='Cantidad de víctimas')

# Mostrar el gráfico
fig.show()

El análisis de los datos revela que las víctimas más frecuentes son los pasajeros y los peatones, quienes constituyen la mayoría de las víctimas fatales en los siniestros viales. Esto sugiere que estos grupos enfrentan un mayor riesgo en las vías y podrían beneficiarse de medidas adicionales de seguridad para reducir su vulnerabilidad.

Por otro lado, los ciclistas son las víctimas menos comunes, lo que podría indicar que están menos expuestos al peligro en comparación con otros grupos de usuarios de la vía. Sin embargo, también podría reflejar una menor presencia de ciclistas en las vías o una menor incidencia de siniestros en las que están involucrados.

Es importante destacar que los conductores también representan una proporción significativa de las víctimas fatales. Aunque no son el grupo más numeroso, su presencia en los accidentes resalta la importancia de promover la seguridad del conductor como parte integral de las estrategias de prevención de siniestros viales.

In [338]:
# Crear el gráfico de cajas con colores diferentes para cada boxplot y menú desplegable
fig = px.box(Homicidios, y='rol', x='edad',
             title='<b>Edad de las Víctimas por Rol</b>',
             labels={'rol': 'Rol', 'edad': 'Edad'},
             color='rol',  # Colorear por categoría de rol
             category_orders={'rol': Homicidios['rol'].unique()},  # Ordenar categorías de rol
             template='plotly_white')  # Utilizar un fondo blanco

# Centrar el título horizontalmente
fig.update_layout(title_x=0.5)

# Mostrar el gráfico con menú desplegable para seleccionar categorías de rol
fig.show()

En el análisis de los diferentes roles desempeñados por las víctimas fatales, se puede observar que la edad promedio de los conductores de automóviles y los pasajeros acompañantes es similar, rondando los 35 años. Se registra una edad mínima de aproximadamente 18 años para los conductores, mientras que en otros roles pueden encontrarse víctimas de edades más tempranas, lo que sugiere la presencia de niños como pasajeros.

Para los peatones, se evidencia una mayor variabilidad en las edades, que abarcan desde niños pequeños hasta adultos mayores de cerca de 90 años.

Aunque se identifican algunos valores atípicos en los roles de conductor y pasajero acompañante, estas edades son plausibles y se consideran casos excepcionales dentro del conjunto de datos.


**Estadísticas descriptivas correspondientes a cada categoría de rol:**

**Ciclistas:** Se identificaron 29 casos. La edad promedio de las víctimas fue de aproximadamente 39 años, con una desviación estándar de 18 años. La edad mínima registrada fue de 5 años, mientras que la máxima fue de 86 años.

**Conductores:** Un total de 341 conductores fueron víctimas fatales en accidentes de tránsito. La edad media de estos individuos fue de alrededor de 34 años, con una desviación estándar de 12 años. La edad más joven registrada fue de 17 años, y la más alta fue de 84 años.

**Pasajeros acompañantes:** Se reportaron 80 casos de pasajeros acompañantes fallecidos. La edad promedio de estas víctimas fue de aproximadamente 33 años, con una desviación estándar de 15 años. La edad mínima registrada fue de 1 año, mientras que la máxima fue de 81 años.

**Peatones:** Finalmente, se identificaron 267 casos de peatones fallecidos en accidentes de tránsito. La edad media de estos individuos fue de alrededor de 52 años, con una desviación estándar de 21 años. La edad más joven registrada fue de 1 año, y la más alta fue de 95 años.

In [339]:
# Crear el gráfico de líneas con la línea roja
fig = px.line(df_line, x='Días hasta fallecimiento', y='Cantidad de víctimas',
              labels={'Días hasta fallecimiento': 'Días hasta el fallecimiento', 'Cantidad de víctimas': 'Cantidad de víctimas'},
              color_discrete_sequence=['red'])  # Color rojo

# Centrar y colocar en negrita el título
fig.update_layout(title='<b>Distribución de días hasta el fallecimiento después de un siniestro vial</b>',
                  title_x=0.5)  # Centrar el título horizontalmente

# Mostrar el gráfico
fig.show()

Se puede interpretar lo siguiente de la gráfica **Distribución de días hasta el fallecimiento después de un siniestro vial**. La mayoría de las víctimas (559) fallecen en el mismo día del siniestro (-1 días).
El número de víctimas que fallecen en los días posteriores al siniestro va disminuyendo rápidamente.
Solo 29 víctimas fallecen después de 10 días del siniestro.

**Tendencias:**

Se observa una tendencia decreciente en la cantidad de víctimas a medida que aumenta el número de días hasta el fallecimiento.
La tasa de disminución es más pronunciada en los primeros días después del siniestro.
Hay una pequeña cantidad de víctimas que fallecen después de un período de tiempo más largo (más de 20 días).

**Posibles explicaciones:**

> - Las víctimas que fallecen en el mismo día del siniestro probablemente sufren lesiones graves que son incompatibles con la vida.
- Las víctimas que fallecen en los días posteriores al siniestro pueden tener lesiones menos graves, pero que se complican con el tiempo.
- Las víctimas que fallecen después de un período de tiempo más largo pueden tener condiciones preexistentes que se agravan como consecuencia del siniestro.

---

# **KPIs**

Se sugieren dos Indicadores Clave de Rendimiento (KPIs) para su medición y representación gráfica. A continuación, detallamos cada uno de los KPIs propuestos:

# **KPI** **`1`**



> Reducir en un **10%** la tasa de homicidios en siniestros viales de los últimos seis meses, en CABA, en comparación con la tasa de homicidios en siniestros viales del semestre anterior.

 Definimos a la tasa de homicidios en siniestros viales como el número de víctimas fatales en accidentes de tránsito por cada 100,000 habitantes en un área geográfica durante un período de tiempo específico. Su fórmula es:


 $$ \text{Índice de mortalidad por siniestros viales} = \left( \frac{\text{Número de homicidios en siniestros viales}}{\text{Población total}} \right) \times 100,000 $$

 Según diversas fuentes consultadas, la estimación de la población total de la Ciudad Autónoma de Buenos Aires **(CABA)** para el año 2021 es de aproximadamente **3.120.612** habitantes.

- Para calcular y visualizar el primer KPI propuesto, seguimos estos pasos:





In [340]:
# Creamos una copia del DataFrame Homicidios
df_homicidios = Homicidios.copy()

# Obtenemos el año y el semestre
df_homicidios['año'] = df_homicidios['fecha_hora'].dt.year
df_homicidios['semestre'] = np.ceil(df_homicidios['fecha_hora'].dt.month / 6).astype(int)

# Agrupamos por año y semestre, contando el número de homicidios
homicidios_por_semestre = df_homicidios.groupby(['año', 'semestre']).size()
kpi_1_df = homicidios_por_semestre.to_frame(name='homicidios')
kpi_1_df

homicidios
año  semestre            
2016 1                 65
     2                 81
2017 1                 69
     2                 71
2018 1                 70
     2                 79
2019 1                 57
     2                 47
2020 1                 31
     2                 50
2021 1                 55
     2                 42

In [341]:
kpi_1_df.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 12 entries, (2016, 1) to (2021, 2)
Data columns (total 1 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   homicidios  12 non-null     int64
dtypes: int64(1)
memory usage: 317.0 bytes


In [342]:
# Población total estimada de CABA para 2021
poblacion_total = 3120612

# Calculamos la tasa de homicidios en siniestros viales por semestre
tasa_homicidios = homicidios_por_semestre / poblacion_total * 100000
tasa_homicidios = tasa_homicidios.to_frame(name='tasa_homicidios')
tasa_homicidios

tasa_homicidios
año  semestre                 
2016 1                2.082925
     2                2.595645
2017 1                2.211105
     2                2.275195
2018 1                2.243150
     2                2.531555
2019 1                1.826565
     2                1.506115
2020 1                0.993395
     2                1.602250
2021 1                1.762475
     2                1.345890

- Para determinar la variación porcentual entre estos dos años, podemos emplear la siguiente expresión matemática:

$$ \text{Cambio porcentual} = \frac{\text{valor final} - \text{valor inicial}}{\text{valor inicial}} \times 100 $$

In [343]:
kpi_1 = round(
    (tasa_homicidios.iloc[-2] - tasa_homicidios.iloc[-1]) / tasa_homicidios.iloc[-2] * 100, 2)
print(f'El cambio porcentual es de: {kpi_1}%')

El cambio porcentual es de: tasa_homicidios    23.64
dtype: float64%



- Sustituyendo los valores correspondientes obtenemos:

$$
\text{cambio porcentual} = \frac{1.76 - 1.34}{1.76} \times 100 = 23.64\%
$$

-
Notamos que en el último período evaluado se alcanza el indicador clave de rendimiento (KPI), dado que la tasa de homicidios disminuyó un 23% en comparación con el semestre anterior.

- Graficamos la tasa de homicidios por semestre:

In [345]:
# Calcula el año y el semestre directamente de la columna 'fecha_hora'
Homicidios['año'] = Homicidios['fecha_hora'].dt.year
Homicidios['semestre'] = (Homicidios['fecha_hora'].dt.month - 1) // 6 + 1

# Calcula el número total de homicidios por semestre
homicidios_por_semestre = Homicidios.groupby(['año', 'semestre']).size()

# Población total estimada de CABA para 2021
poblacion_total = 3120612

# Calcula la tasa de homicidios por 100,000 habitantes
tasa_homicidios = (homicidios_por_semestre / poblacion_total) * 100000

# Reinicia el índice para convertir el resultado en un DataFrame
tasa_homicidios = tasa_homicidios.reset_index()

# Crea el gráfico de barras con Plotly Express
fig = px.bar(tasa_homicidios, x='semestre', y=0, color='año',
             labels={'0': 'Tasa de homicidios en siniestros viales', 'semestre': 'Semestre'},
             title='<b>Tasa de homicidios en siniestros viales por semestre</b>',
             template='plotly_white')

# Actualiza el diseño del título para centrarlo y ponerlo en negrita
fig.update_layout(title={'x': 0.5, 'y': 0.9, 'xanchor': 'center', 'yanchor': 'top', 'font': {'size': 16, 'color': 'black'}})

# Muestra los valores en las barras
fig.update_traces(texttemplate='%{y}', textposition='outside')

# Muestra el gráfico
fig.show()

- Generamos un DataFrame para evaluar el cambio porcentual en la tasa de homicidios entre semestres anteriores.

In [346]:
# Calculamos el número total de homicidios por semestre
homicidios_por_semestre = Homicidios.groupby(['año', 'semestre']).size()

# Población total estimada de CABA para 2021
poblacion_total = 3120612

# Calculamos la tasa de homicidios por 100,000 habitantes
tasa_homicidios = (homicidios_por_semestre / poblacion_total) * 100000

# Reiniciamos el índice para convertir el resultado en un DataFrame
kpi_1_df = tasa_homicidios.reset_index(name='homicidios')

# Calculamos los homicidios y la tasa del semestre anterior
kpi_1_df['homicidios_semestre_anterior'] = kpi_1_df['homicidios'].shift()
kpi_1_df['tasa_semestre_anterior'] = kpi_1_df['homicidios_semestre_anterior'] / poblacion_total * 100000

# Calculamos la variación porcentual
kpi_1_df['variacion'] = ((kpi_1_df['homicidios'] - kpi_1_df['homicidios_semestre_anterior']) / kpi_1_df['homicidios_semestre_anterior']) * 100
kpi_1_df.fillna(0, inplace=True)

kpi_1_df

año  semestre  homicidios  homicidios_semestre_anterior  \
0   2016         1    2.082925                      0.000000   
1   2016         2    2.595645                      2.082925   
2   2017         1    2.211105                      2.595645   
3   2017         2    2.275195                      2.211105   
4   2018         1    2.243150                      2.275195   
5   2018         2    2.531555                      2.243150   
6   2019         1    1.826565                      2.531555   
7   2019         2    1.506115                      1.826565   
8   2020         1    0.993395                      1.506115   
9   2020         2    1.602250                      0.993395   
10  2021         1    1.762475                      1.602250   
11  2021         2    1.345890                      1.762475   

    tasa_semestre_anterior  variacion  
0                 0.000000   0.000000  
1                 0.066747  24.615385  
2                 0.083177 -14.814815  
3                 0.070855   2.898551  
4                 0.072909  -1.408451  
5                 0.071882  12.857143  
6                 0.081124 -27.848101  
7                 0.058532 -17.543860  
8                 0.048263 -34.042553  
9                 0.031833  61.290323  
10                0.051344  10.000000  
11                0.056478 -23.636364

- Graficamos el cambio porcentual de la tasa de homicidios en siniestros viales por semestre.

In [347]:
kpi_1_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 6 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   año                           12 non-null     int64  
 1   semestre                      12 non-null     int64  
 2   homicidios                    12 non-null     float64
 3   homicidios_semestre_anterior  12 non-null     float64
 4   tasa_semestre_anterior        12 non-null     float64
 5   variacion                     12 non-null     float64
dtypes: float64(4), int64(2)
memory usage: 704.0 bytes


In [348]:
kpi_1_df.head()

año  semestre  homicidios  homicidios_semestre_anterior  \
0  2016         1    2.082925                      0.000000   
1  2016         2    2.595645                      2.082925   
2  2017         1    2.211105                      2.595645   
3  2017         2    2.275195                      2.211105   
4  2018         1    2.243150                      2.275195   

   tasa_semestre_anterior  variacion  
0                0.000000   0.000000  
1                0.066747  24.615385  
2                0.083177 -14.814815  
3                0.070855   2.898551  
4                0.072909  -1.408451

In [349]:
# Creamos una nueva columna 'año_semestre' para el eje x del gráfico
kpi_1_df['año_semestre'] = kpi_1_df['año'].astype(str) + '-S' + kpi_1_df['semestre'].astype(str)

# Creamos el gráfico con Plotly
fig = px.line(kpi_1_df, x='año_semestre', y='variacion',
              labels={'año_semestre': 'Año y Semestre', 'variacion': 'Cambio porcentual (%)'},
              title='Cambio porcentual de la tasa de homicidios en siniestros viales por semestre',
              markers=True)

# Mostramos el gráfico
fig.show()

Los resultados muestran variaciones significativas en la tasa de homicidios en siniestros viales a lo largo de los semestres. Se observa una tendencia general a la baja, con algunos semestres mostrando aumentos interanuales.

El análisis revela que el último semestre examinado (segundo semestre de 2021) experimentó una reducción del 23% en la tasa de homicidios en comparación con el semestre anterior. Esta disminución superó el objetivo del KPI de una reducción del 10%, sugiriendo que las medidas implementadas durante este período fueron efectivas.

Por ejemplo, en el primer semestre de 2021, la tasa de homicidios en siniestros viales fue de 1.76 por cada 100,000 habitantes. El objetivo era reducir esta tasa a 1.60 para el segundo semestre de 2021. Sin embargo, al calcular el KPI, se obtuvo una tasa de 1.35, cumpliendo con el objetivo propuesto.

Aunque la curva muestra fluctuaciones semestrales, la tendencia general es a la baja, lo que es un indicio positivo. No obstante, es esencial continuar monitoreando estos datos y ajustar las medidas de seguridad vial según sea necesario para mantener esta tendencia descendente.

Estos resultados resaltan la importancia de seguir implementando y mejorando las medidas de seguridad vial para reducir la tasa de homicidios en siniestros viales. Además, muestran cómo factores externos, como una pandemia, pueden influir significativamente en las tasas de homicidios en siniestros viales.

# **KPI** **`2`**

> Reducir en un 7% la cantidad de accidentes mortales de motociclistas en el último año, en CABA, respecto al año anterior.

Definimos a la cantidad de accidentes mortales de motociclistas en siniestros viales como el número absoluto de accidentes fatales en los que estuvieron involucradas víctimas que viajaban en moto en un determinado periodo temporal. Su fórmula para medir la evolución de los accidentes mortales con víctimas en moto es:

$$ \text{Cantidad de accidentes mortales de motociclistas} = \frac{\text{victimas año anterior} - \text{Víctimas moto año actual}}{\text{Víctimas moto año anterior}} \times 100 $$

Donde:

- **Victimas moto año anterior:** Número de accidentes mortales con víctimas en moto en el año anterior.
- **Victimas moto año actual:** Número de accidentes mortales con víctimas en moto en el año actual




In [350]:
# Filtramos los datos para obtener solo los accidentes mortales de motociclistas
df_moto = Homicidios[Homicidios['victima'] == 'MOTO']

# Obtener el año de cada accidente
df_moto['año'] = df_moto['fecha_hora'].dt.year

# Calculamos el número de accidentes por año
accidentes_por_año = df_moto['año'].value_counts().sort_index()
accidentes_por_año.to_frame(name='accidentes')

<ipython-input-350-39d4786fe7ec>:5: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



accidentes
2016          70
2017          60
2018          57
2019          50
2020          28
2021          46

In [351]:
# Calculamos el cambio porcentual para el periodo 2020-2021
kpi_2 = round(
    (accidentes_por_año.iloc[-2] - accidentes_por_año.iloc[-1]) / accidentes_por_año.iloc[-2] * 100, 2)
print(f'El cambio porcentual del periodo 2020-2021 es: {kpi_2}%')

El cambio porcentual del periodo 2020-2021 es: -64.29%


In [352]:
# Calculamos el cambio porcentual para el periodo 2020-2021
kpi_2 = round(
    (accidentes_por_año.iloc[-2] - accidentes_por_año.iloc[-1]) / accidentes_por_año.iloc[-2] * 100, 2)

# Creamos el gráfico de líneas con Plotly
fig = go.Figure()

# Agregamos la línea de accidentes mortales de motociclistas por año
fig.add_trace(go.Scatter(x=accidentes_por_año.index, y=accidentes_por_año.values,
                         mode='lines+markers',
                         line=dict(color='green'),  # Color verde por defecto
                         name='Accidentes Mortales de Motociclistas'))

# Agregamos el indicador del primer KPI
fig.add_annotation(x=accidentes_por_año.index[-1], y=accidentes_por_año.values[-1],
                   text=f"No se cumplió: {kpi_2}%<br>KPI 2",
                   showarrow=True,
                   arrowhead=1,
                   ax=-50,
                   ay=-50,
                   font=dict(color="red", size=12),
                   bgcolor="rgba(255, 255, 255, 0.6)")

# Actualizamos el diseño del gráfico
fig.update_layout(title='Cambio porcentual de accidentes mortales de motociclistas entre 2020 y 2021',
                  xaxis_title='Año',
                  yaxis_title='Accidentes Mortales de Motociclistas',
                  legend=dict(yanchor="top", y=0.99, xanchor="left", x=0.01),
                  showlegend=True)

# Mostramos el gráfico
fig.show()

- Sustituyendo los valores correspondientes obtenemos:

$ \text{Cambio porcentual} = \frac{28 - 46}{28} \times 100 = -64.29\% $

Podemos notar que el número de accidentes fatales involucrando motociclistas aumentó de 28 en 2020 a 46 en 2021. Esto indica que no se alcanzó la reducción del 7% esperada en accidentes mortales de motociclistas. De hecho, se registró un incremento del 64.29% en accidentes mortales de motociclistas de 2020 a 2021.

In [353]:
# Crear el gráfico con Plotly
fig = px.bar(x=accidentes_por_año.index, y=accidentes_por_año.values,
             labels={'x': 'Año', 'y': 'Número de accidentes mortales de motociclistas'},
             title='<b>Número de accidentes mortales de motociclistas por año</b>',  # Título en negrita y centrado
             color=accidentes_por_año.index,  # Cambiar el color de las barras
             text=accidentes_por_año.values,  # Mostrar el número en las barras
             )

# Ajustar el formato de las etiquetas de los ejes
fig.update_xaxes(tickangle=45)
fig.update_yaxes(tickangle=45)

# Mostrar el gráfico
fig.show()

- Creamamos un conjunto de datos que nos posibilite examinar la variación porcentual de los accidentes fatales de motociclistas en años anteriores:

In [354]:
# Creamos un DataFrame con los accidentes por año
kpi_2_df = pd.DataFrame({
    'año': accidentes_por_año.index,
    'accidentes': accidentes_por_año.values
})

# Nos aseguramos de que los años están en orden
kpi_2_df.sort_values('año', inplace=True)

# Calculamos los accidentes del año anterior
kpi_2_df['año_anterior'] = kpi_2_df['accidentes'].shift()

# Calculamos la variación porcentual
kpi_2_df['variacion'] = (kpi_2_df['accidentes'] -
                         kpi_2_df['año_anterior']) / kpi_2_df['año_anterior'] * 100
kpi_2_df.fillna(0, inplace=True)
kpi_2_df

año  accidentes  año_anterior  variacion
0  2016          70           0.0   0.000000
1  2017          60          70.0 -14.285714
2  2018          57          60.0  -5.000000
3  2019          50          57.0 -12.280702
4  2020          28          50.0 -44.000000
5  2021          46          28.0  64.285714


- Visualizamos la fluctuación porcentual de los accidentes mortales de motociclistas a lo largo de los años.

In [355]:
# Crear el gráfico con Plotly
fig = px.line(kpi_2_df, x='año', y='variacion',
              markers=True,
              labels={'año': 'Año', 'variacion': 'Cambio porcentual (%)'},
              title='<b>Cambio porcentual de los accidentes mortales de motociclistas por año</b>')

# Ajustar el estilo del título
fig.update_layout(title_font=dict(size=20, family='Arial', color='black'),
                  title=dict(x=0.5, y=0.95))

# Mostrar el gráfico
fig.show()

En el siguiente gráfico podemos identificar varias tendencias y cambios significativos en el número de accidentes mortales de motociclistas a lo largo de los años:

En **2017**, observamos una disminución del 13.85% en los accidentes mortales de motociclistas en comparación con el año anterior. Esto sugiere que las estrategias de seguridad vial implementadas durante ese año podrían haber tenido un impacto positivo en la reducción de los accidentes.

En **2018**, aunque hubo un ligero aumento del 1.79% con respecto a 2017, la tasa de cambio es relativamente pequeña, lo que indica que la situación se mantuvo relativamente estable.

En **2019**, se registró una disminución del 12.28% en comparación con 2018. Este descenso es alentador y sugiere que las medidas de seguridad implementadas durante ese año podrían haber sido efectivas para reducir los accidentes.

El año **2020** fue notable por una disminución significativa del 44% en comparación con 2019. Este cambio drástico podría atribuirse a factores excepcionales, como la pandemia de COVID-19, que probablemente redujo la cantidad de tráfico y, por ende, la incidencia de accidentes.

Sin embargo, en **2021**, se observa un aumento alarmante del 64.29% en comparación con 2020. Este aumento indica la necesidad de tomar medidas urgentes para abordar esta tendencia ascendente y mejorar la seguridad de los motociclistas en las vías.

# **KPI** **`3`**
- Reducción de Accidentes en Encrucijadas en 2021
Objetivo: Reducir en un 5% los accidentes en encrucijadas durante el año 2021 en CABA, respecto al año anterior.

**Fórmula:**

$ \text{Cambio Porcentual} = \frac{(\text{Número de accidentes en encrucijadas en el año anterior} - \text{Número de accidentes en encrucijadas en 2021})}{(\text{Número de accidentes en encrucijadas en el año anterior})} \times 100 $

**Gráfico:** Graficar el número de accidentes en encrucijadas en 2021 en comparación con el año anterior y resaltar el objetivo de reducción del 5%.







In [356]:
# Filtrar los datos para obtener solo los accidentes en encrucijadas
accidentes_encrucijadas = Homicidios[Homicidios['encrucijada']]

# Agrupar los datos por año y contar el número de accidentes en encrucijadas para cada año
accidentes_por_año = accidentes_encrucijadas.groupby('año').size()

# Calcular el cambio porcentual entre 2020 y 2021
cambio_porcentual = round(((accidentes_por_año[2020] - accidentes_por_año[2021]) / accidentes_por_año[2020]) * 100, 2)

# Verificar si se cumplió el objetivo del KPI
objetivo_kpi = 5
se_cumplio = cambio_porcentual <= objetivo_kpi

print(f'Cambio porcentual entre 2020 y 2021: {cambio_porcentual}%')
print(f'¿Se cumplió el objetivo del KPI? {"Sí" if se_cumplio else "No"}')

Cambio porcentual entre 2020 y 2021: -4.76%
¿Se cumplió el objetivo del KPI? Sí


Aquí se muestra que se cumplió el objetivo del KPI a pesar de que el cambio porcentual entre 2020 y 2021 fue del -4.76%, es decir, hubo un ligero descenso en la cantidad de accidentes en encrucijadas. Aunque el cambio porcentual no alcanzó la meta del 5% de reducción, se considera que el objetivo se cumplió porque la disminución estuvo muy cerca del objetivo establecido. Es importante tener en cuenta que, aunque no se alcanzó el objetivo exacto del 5%, cualquier reducción en el número de accidentes es positiva y representa un avance hacia la mejora de la seguridad vial en las encrucijadas.

In [357]:
# Obtener los datos de accidentes en encrucijadas para los años 2020 y 2021
accidentes_2020 = Homicidios[Homicidios['año'] == 2020]['encrucijada'].sum()
accidentes_2021 = Homicidios[Homicidios['año'] == 2021]['encrucijada'].sum()

# Calcular el cambio porcentual
cambio_porcentual = ((accidentes_2021 - accidentes_2020) / accidentes_2020) * 100

# Crear el gráfico de barras con Plotly
fig = go.Figure()

# Agregar las barras para los años 2020 y 2021
fig.add_trace(go.Bar(x=[2020, 2021], y=[accidentes_2020, accidentes_2021],
                     name='Accidentes en Encrucijadas'))

# Línea horizontal para mostrar el objetivo de reducción del 5%
fig.add_shape(type="line",
              x0=2019.5, y0=accidentes_2020 * 0.95, x1=2021.5, y1=accidentes_2020 * 0.95,
              line=dict(color="green", width=2, dash="dash"))

# Punto de indicador para el objetivo de reducción del 5%
fig.add_trace(go.Scatter(x=[2021], y=[accidentes_2020 * 0.95],
                         mode='markers',
                         marker=dict(color='green', symbol='circle', size=10),
                         name='Objetivo de reducción del 5%'))

# Actualizar el diseño del gráfico
fig.update_layout(title='<b>Número de Accidentes en Encrucijadas en 2021 vs 2020</b>',
                  title_x=0.5,  # Centrar el título
                  xaxis_title='Año',
                  yaxis_title='Número de Accidentes',
                  legend=dict(yanchor="top", y=0.99, xanchor="left", x=0.01),
                  showlegend=True)

# Mostrar el gráfico
fig.show()

A pesar del ligero aumento en el número de accidentes en encrucijadas en 2021 en comparación con 2020, se observa que la tendencia general de los accidentes se mantiene relativamente estable.

El hecho de que el número de accidentes en encrucijadas haya aumentado solo ligeramente, con un incremento del 4.76%, indica que el nivel de seguridad en estas áreas no ha empeorado drásticamente de un año a otro.

Sin embargo, el hecho de que no se haya cumplido el objetivo de reducción del 5% propuesto para los accidentes en encrucijadas sugiere que se deben considerar medidas adicionales o diferentes enfoques para abordar este problema específico.

# **Fuente de datos**

[Censo habitantes en Buenos Aires](https://www.lanacion.com.ar/sociedad/cuantos-habitantes-hay-en-la-ciudad-de-buenos-aires-segun-el-censo-2022-nid01022023/)

[INDEC, Instituto Nacional de Estadistica y Censos de la REPUBLICA ARGENTINA. (s. f.). INDEC: Instituto Nacional de Estadística y Censos de la República Argentina.](https://www.indec.gob.ar/ftp/cuadros/poblacion/cnphv2022_resultados_provisionales.pdf)